pkg/network/protocols/http/event_pipeline_test.go

```diff
@@ -0,0 +1,173 @@
+		mockBatch.Idx = uint64(i)
+		var buf bytes.Buffer
+		require.NoError(b, binary.Write(&buf, binary.LittleEndian, &mockBatch))
+		events.RecordSample(c, consumer, buf.Bytes())
+		buf.Reset()
+	}
+}
+
+// createHTTPEvents creates a slice of HTTP events to be used in the benchmark.
+func createHTTPEvents(eventsData []HTTPEventData) []EbpfEvent {
+	events := make([]EbpfEvent, len(eventsData))
+	for i, data := range eventsData {
+		events[i] = EbpfEvent{
+			Tuple: ConnTuple{},
+			Http: EbpfTx{
+				Request_started:      1,
+				Response_last_seen:   2,
+				Request_method:       data.Method,
+				Response_status_code: data.StatusCode,
+				Request_fragment:     requestFragment(data.RequestFragment),
+			},
+		}
+	}
+	return events
+}
+
+// BenchmarkHTTPEventConsumer benchmarks the consumer with a large number of events to measure the performance.
+func BenchmarkHTTPEventConsumer(b *testing.B) {
+	// Set MemProfileRate to 1 in order to collect every allocation
+	runtime.MemProfileRate = 1
+	var wg sync.WaitGroup
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	testCases := []struct {
+		name             string
+		totalEventsCount int
+		// Serialized data can't exceed batchDataSize bytes that why we can insert 14 events in a batch.
+		numOfEventsInBatch int
+		httpEvents         []EbpfEvent
+	}{
+		{"SmallBatch", 1000, 8, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"MediumBatch", 38000, 10, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"LargeBatch", 42000, 14, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"MaxEventsCount", 3150000, 14, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodDelete), StatusCode: http.StatusAccepted, RequestFragment: []byte("DELETE /delete HTTP/1.1")},
+		})},
+	}
+
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				consumer, p := setupBenchmark(b, config.New(), tc.totalEventsCount, tc.numOfEventsInBatch, tc.httpEvents, &wg)
+
+				consumer.Start()
+				require.Eventually(b, func() bool {
+					if tc.totalEventsCount == int(p.telemetry.hits2XX.counterPlain.Get()) {
+						b.Logf("USM summary: %s", p.telemetry.metricGroup.Summary())
+						p.telemetry.hits2XX.counterPlain.Reset()
```

Comment:
This seems redundant as you create a new protocol for each iteration

Reply:
removed the function

Reply:
I used a unique name when creating the telemetry to avoid creating metric groups with the same name. Another possible solution I found is using telemetry.Reset(), as done in other tests.